File: clients/CSharp/Model/RepairAction.cs
namespace Aicup2020.Model
{
    public struct RepairAction
    {
        public int Target { get; set; }
        public RepairAction(int target)
        {
            this.Target = target;
        }
        public static RepairAction ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new RepairAction();
            result.Target = reader.ReadInt32();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(Target);
        }
    }
}

File: clients/FSharp/Model/RepairAction.fs
#nowarn "0058"
namespace Aicup2020.Model
type RepairAction = {
    Target: int;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.Target
    static member readFrom(reader: System.IO.BinaryReader) = {
        Target = reader.ReadInt32()
    }
